websocket.go

```diff
@@ -652,7 +652,8 @@ func (c *wsConn) handleWsConn(ctx context.Context) {
 			}
 			c.writeLk.Unlock()
 			log.Errorw("Connection timeout", "remote", c.conn.RemoteAddr())
-			return
+			// Maybe the server was just shut down for a while, and then it started up again.
+			continue
```

Comment:
Are you sure that c.incoming will actually not hang / return an error (and make us reconnect) when we close the connection here? (even in cases when the server looses connection, then reboots, with no chance to cleanly close the TCP connection)

This is why this is was a return instead of a continue -> https://github.com/filecoin-project/go-jsonrpc/pull/23#discussion_r471565640

Reply:
I think the problem is that both the client side and the server side use `handleWsConn`, but their strategies for handling connections should be slightly different

The client will only start handleWsConn once when NewClient called, if handleWsConn exit the client instance will become unavailable, rather than external NewClient again to get a new Client instance, but we should be looking forward to don't need to obtain a new instance Client again, it should be able to automatically reconnect, so client processing handleWsConn should have a principle: don't leave the function, but keep trying reconnection

The server starts a handleWsConn at ServeHTTP, and the client reconnects with a handleWsConn at every disconnect, so we need it to exit correctly, so the principle of handling handleWsConn on the server side is to close the connection and exit when the current connection is abnormal

Moreover, only the client has assigned a value of `connfactory`. Therefore, we can judge whether handlewsconn is called by the client or by the server, and implement the above two principles at a time